inc/media.php

```diff
@@ -1697,15 +1697,17 @@ function media_nstree($ns)
         else $tmp_ns = $part;
 
         // find the namespace parts or insert them
-        while ($data[$pos]['id'] != $tmp_ns) {
-            if (
-                $pos >= count($data) ||
-                ($data[$pos]['level'] <= $level + 1 && Sort::strcmp($data[$pos]['id'], $tmp_ns) > 0)
-            ) {
-                array_splice($data, $pos, 0, [['level' => $level + 1, 'id' => $tmp_ns, 'open' => 'true']]);
-                break;
+        if (!empty($data[$pos]['id'])) {
+            while ($data[$pos]['id'] != $tmp_ns) {
```

Comment:
I'm just looking at this small piece of code without taking into consideration the greater environment. But wouldn't it be better to check on each loop iteration? I.e.:
```php
        while (isset($data[$pos]['id']) && !empty($data[$pos]['id']) && $data[$pos]['id'] != $tmp_ns) {
            if (
                $pos >= count($data) ||
                ($data[$pos]['level'] <= $level + 1 && Sort::strcmp($data[$pos]['id'], $tmp_ns) > 0)
            ) {
                array_splice($data, $pos, 0, [['level' => $level + 1, 'id' => $tmp_ns, 'open' => 'true']]);
                break;
            }
            ++$pos;
        }
```
Checking only at the the start with `$pos === 0` ignores the potential cases where `$data[$pos]['id']` does not exist or is empty for `$pos > 0` values. I figure that if `$data[0]['id']` can be empty, others can be as well?

And the `isset()` test ensures that the array element exists at all. But if it can be proven that it always exists then this check can be removed.